doc/developer/design/20230613_cloudtest_improvements.md

```diff
@@ -0,0 +1,101 @@
+# Cloudtest improvements
+
+Associated:
+* https://github.com/MaterializeInc/materialize/issues/19626
+
+## Summary
+
+The materialize repository and cloud repository both contain their own collection of cloudtests. These tests orchestrate
+cloud resources and verify the behavior of Materialize running in the cloud. However, their setup is different and does
+not share a common base.
+
+This document proposes an improvement to these cloudtests by unifying their test setup across both repositories and
+sharing existing test framework functionality.
+
+## Goals & Non-Goals
+
+### Goals
+* Have a single, unified setup and test framework for cloudtests in both repos, thereby
+  * reduce efforts to get familiar with both code bases
+  * reduce code duplication and resulting maintenance efforts
+* Provide all relevant test framework functionalities to both repositories
+* Lay the ground for future improvements
+
+### Non-Goals
+* Rework all existing cloudtests
+
+## Description
+
+The unified test framework shall be located in the materialize repo to allow access from both repos. It will be present
+in the cloud repo through the inclusion of the materialize repo as git submodule. This submodule is synchronized
+occasionally (usually as part of the release process).
+
+The existing cloudtest framework in the materialize repo provides an instance of class `MaterializeApplication` as
+context to each test case. This class inherits `Application` and brings functionality to access involved cloud
+resources and interact with kubernetes control.
+
+A further class `CloudtestApplication` inheriting `Application` shall be introduced. It will expose existing
+functionality from materialize repo cloudtests. In addition, it will provide further utility functions to interact
+with the controllers of the cloud repo. Where necessary and sensible, functionality from `MaterializeApplication`
+shall be pulled up into the shared base class.
+
+A few sample tests using the unified framework will be created in the cloud repo. Their execution shall be integrated
+into the CI.
+
+## Drawbacks
+
+* Changing the test framework, which resides in the materialize repo, might break test execution in cloud repo; in
+particular when a developer is not aware that the logic is used there as well. This will be mitigated to some degree
+by the build which is executed when synchronizing the materialize repo to the cloud repo.
```

Comment:
The plan is to run tests from cloud repository in the cloud's CI, which uses GitHub actions.

While it might be possible to run them in materialize repo's CI as well, it would require Buildkite to check out another repository and, more importantly, could leak data from the cloud repository, which is private.